onadata/apps/api/viewsets/open_id_connect_viewset.py

```diff
@@ -108,7 +108,9 @@ def callback(self, request, **kwargs):  # pylint: disable=no-self-use
                               "Please choose a different one.")
                 data = {'error_msg': error_msg, 'id_token': id_token}
             else:
-                email = decoded_token.get('email')
+                claim_values = oidc_handler.get_claim_values(
+                    ['email', 'given_name', 'family_name'], decoded_token)
```

Comment:
Could we have all the values, `'email', 'given_name', 'family_name'`, configurable from static or settings?

```python
[EMAIL, GIVEN_NAME, FAMILY_NAME]
```
The fields could be different with different providers.

Reply:
They are currently configurable in the settings within the `OPEN_ID_CONNECT_PROVIDERS` setting. A user can pass in a `claims` key in a providers configuration and set custom claim values to retrieve, like:

```
'msft': { 'claims': 'email': 'sub'}
```

Reply:
In this case the email claim will either be the default `email` if the config doesn't have an `email` claim_name value set or `sub` in the case above

Reply:
I would still recommend we have a constant for the 3 values `[EMAIL, GIVEN_NAME, FAMILY_NAME]`.

Reply:
Would this be on top of the current change or should we implement the constants only and remove the `claims` key on the `OPEN_ID_CONNECT_PROVIDERS`

Reply:
We have some constants in [onadata/libs/utils/common_tags.py](https://github.com/onaio/onadata/blob/master/onadata/libs/utils/common_tags.py). We would use them even if we use the `claims` in settings.

I, however, see `claims` and the values used are part of OpenID standard - https://openid.net/specs/openid-connect-core-1_0.html#Claims - but one way to minimise on the potential for spelling mistakes would be to use constants.

Reply:
Added the common tags, the settings now act as an override for the email, first name and last name claim. One can override the `EMAIL_CLAIM` by setting:

```
'claims': {
        EMAIL_CLAIM: SUBJECT_CLAIM
}
```

Reply:
I was thinking more of

 ```python
        'claims': {
            EMAIL: 'sub',
            GIVEN_NAME: 'fname',
            FAMILY_NAME: 'lname',
        },
```

Then
```python
from onadata.libs.utils.common_tags import EMAIL, FAMILY_NAME, GIVEN_NAME
.
..
...
..
.
                claim_values = oidc_handler.get_claim_values(
                    [EMAIL, GIVEN_NAME, FAMILY_NAME],
                    decoded_token)
                email = claim_values.get(EMAIL)
...
```

Reply:
Implemented this change in the latest commits